shell/exercism_completion.zsh

```diff
@@ -33,4 +35,4 @@ _exercism() {
     esac
 }
 
-compdef '_exercism' exercism
+_exercism "$@"
```

Comment:
This is the wrong way.

If this file is put into the `fpath`, then its already the function `_exercism`, then you define another internal scoped funtion with the same name and unconditionally call that.

This gives us an extra level of calling.

The proper way is to remove the enclosing function definition and remove this last line.

Thats the theory at least.

Problem with this though, as far as I remember, these files are autogenerated by the CLI-Framework.

Reply:
Of the three that I looked at, two used a function, but you're right that the majority of zsh completions I have installed do not have the extra function.

If the file isn't autogenerated, then I can certainly go that way.